obsreport/obsreporttest/obsreporttest.go

```diff
@@ -41,17 +45,35 @@ var (
 	processorTag, _ = tag.NewKey("processor")
 )
 
+type TestTelemetrySettings struct {
+	component.TelemetrySettings
+	SpanRecorder           *tracetest.SpanRecorder
+	ExporterCreateSettings component.ExporterCreateSettings
+}
+
 // SetupRecordedMetricsTest does setup the testing environment to check the metrics recorded by receivers, producers or exporters.
 // The returned function should be deferred.
-func SetupRecordedMetricsTest() (func(), error) {
+func SetupRecordedMetricsTest() (TestTelemetrySettings, func(), error) {
```

Comment:
In an ideal world the "done" func will not exist since the otel metrics will not be a singleton. How can we design this?

Reply:
@codeboten what do you think about having a "Shutdown" on the `TestTelemetrySettings` that will shutdown the providers/flush logs/ and for the moment uninstall opencensus views? Do you think this is an ok path?

Reply:
👍 i like it, makes it cleaner than the return function.